pageserver/src/tenant/mgr.rs

```diff
@@ -537,7 +646,9 @@ pub async fn load_tenant(
             broker_client,
             remote_storage,
         };
-        let new_tenant = schedule_local_tenant_processing(conf, &tenant_path,  resources, None,  &TENANTS, ctx)
+        // TODO: remove the `/load` API once generation support is complete:
+        // it becomes equivalent to attaching.
+        let new_tenant = schedule_local_tenant_processing(conf, tenant_id, &tenant_path, Generation::none(), resources, None,  &TENANTS, ctx)
             .with_context(|| {
```

Comment:
Hm, when reading this, I realized we don't store `Generation` durably on disk (for good reason, there's no scenario in which it would be used correctly.)

But, it means that `/ignore` and `/load` are unsafe to use if generations are enabled.


IMO we should mnake it abundantly clear in the code that these APIs are unsafe in a world with generations.

Also, make this decision earlier, outside of `tenant_map_insert`, so that we don't remove the `ignore` mark too early.

```rust
let generation = match config.control_plane_api {
    Some(_) => anyhow::bail!("/ignore and /load are unsafe in a world with generation numbers"),
    None => Generation::none(),
};
```

Reply:
https://github.com/neondatabase/neon/pull/5163/commits/891f7c4ccc78505f9d7ea255ca30787e62f68852 sovles this, right? Please resolve if that's the case

Reply:
Yes, that fixes the safety question for /load.

In some later PR I will still make the generation persistent on disk, because it's useful for the deletion queue optimization where we can validate a previous generation if it was the last one attached to this  server, and because it will be useful for our "escape hatch" to do our best to run without a control plane to give us a generation.